src/Chapter3.hs

```diff
@@ -480,6 +514,34 @@ After defining the city, implement the following functions:
    and at least 10 living __people__ inside in all houses of the city totally.
 -}
 
+data OptionalCastle 
+  = NoCastle
+  | Castle 
+    { name :: String
+    , wall :: Bool} deriving (Show)
+
+data House = MkHouse {
+  housePeople :: Integer
```

Comment:
Interesting idea to have `Integer` here.
However, in this case, nothing stops us from creating houses with any amount of people (even negative).

One approach for this is to create an additional function and allow to create `House` value only with it, and restrict constructor usage. It is usually called "smart constructor". To avoid building houses with more than 4 people you need to restrict usage of the Constructor and allow to create `House`s only with your `mkHouse` function.
You can read a bit more on the pattern in here: https://kowainik.github.io/posts/haskell-mini-patterns#smart-constructor

Generally, in here, you could restrict that on type level, by using something else than `Integer` 🙂